apis/Google.Cloud.Dialogflow.V2/Google.Cloud.Dialogflow.V2.Snippets/AgentsClientSnippets.cs

```diff
@@ -48,5 +52,37 @@ public void SearchAgents()
             }
             Console.WriteLine($"Total agents: {allAgents.Count}");
         }
+
+        // Sample: Webhook
+        public class DialogflowController : ControllerBase
+        {
+            // A Protobuf JSON parser configured to ignore unknown fields. This makes
+            // the action robust against new fields being introduced by Dialogflow.
+            private static readonly JsonParser jsonParser =
+                new JsonParser(JsonParser.Settings.Default.WithIgnoreUnknownFields(true));
+
+            public ContentResult DialogAction()
+            {
+                // Parse the body of the request using the Protobuf JSON parser,
+                // *not* Json.NET.
+                WebhookRequest request;
+                using (var reader = new StreamReader(Request.Body))
+                {
+                    request = jsonParser.Parse<WebhookRequest>(reader);
+                }
+
+                // Note: you should authenticate the request here.
+
+                // Populate the response
+                WebhookResponse response = new WebhookResponse
+                {
+                    // ...
+                };
+
+                // Return the response, using Protobuf JSON formatting, *not* Json.NET.
```

Comment:
nit: I assume the `.ToString()` call is performing JSON formatting using the protobuf JSON formatter? It may be worth stating this more clearly in this comment.

Reply:
Yes. Will do.